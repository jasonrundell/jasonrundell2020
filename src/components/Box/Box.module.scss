.root {
  padding: $normalUnit;
}

.tight {
  padding: $smallUnit;
}

.roomy {
  padding: $xxLargeUnit;
}
