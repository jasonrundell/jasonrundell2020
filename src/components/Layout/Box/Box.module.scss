@import '../../../../src/styles/main.scss';

.root {
  padding: $sizeNormal;
  line-height: $sizeNormal;
}

.tight {
  padding: $sizeSmallest;
}

.roomy {
  padding: $sizeLargest;
}
