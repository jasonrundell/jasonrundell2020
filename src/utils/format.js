/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
