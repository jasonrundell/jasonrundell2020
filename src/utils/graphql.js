export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
