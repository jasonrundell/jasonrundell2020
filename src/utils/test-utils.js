export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
