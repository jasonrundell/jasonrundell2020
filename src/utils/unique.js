const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
