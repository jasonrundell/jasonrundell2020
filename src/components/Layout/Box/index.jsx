import React from 'react'
import PropTypes from 'prop-types'

import { classNames } from '../../../utils/css'

import styles from './Box.module.scss'

const Box = ({ children, isTight, isRoomy }) => {
  const className = classNames(
    styles.root,
    isTight && styles.tight,
    isRoomy && styles.roomy
  )
  return <div className={className}>{children}</div>
}

Box.defaultProps = {
  isTight: false,
  isRoomy: false,
}

Box.propTypes = {
  isTight: PropTypes.bool,
  isRoomy: PropTypes.bool,
  children: PropTypes.any.isRequired,
}

export default Box
