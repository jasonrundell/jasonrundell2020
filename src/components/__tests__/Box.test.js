import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Box from '../Layout/Box'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Box component', () => {
  it('Box renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <Box>Box component</Box>
          <Box isTight>isTight true</Box>
          <Box isTight={false}>isTight false</Box>
          <Box isRoomy>isRoomy true</Box>
          <Box isRoomy={false}>isRoomy false</Box>
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Box has required children prop', () => {
    act(() => {
      render(<Box>Box component</Box>, container)
    })
    expect(container.textContent).toBe('Box component')
  })

  it('Box isTight prop', () => {
    act(() => {
      render(<Box isTight={true}>isTight true</Box>, container)
    })
    expect(container.textContent).toBe('isTight true')

    act(() => {
      render(<Box isTight={false}>isTight false</Box>, container)
    })
    expect(container.textContent).toBe('isTight false')
  })

  it('Box isRoomy prop', () => {
    act(() => {
      render(<Box isRoomy={true}>isRoomy true</Box>, container)
    })
    expect(container.textContent).toBe('isRoomy true')

    act(() => {
      render(<Box isRoomy={false}>isRoomy false</Box>, container)
    })
    expect(container.textContent).toBe('isRoomy false')
  })
})
