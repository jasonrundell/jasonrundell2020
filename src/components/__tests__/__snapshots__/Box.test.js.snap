// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component Box renders correctly 1`] = `
<div>
  <div
    className="root"
  >
    Box component
  </div>
  <div
    className="root tight"
  >
    isTight true
  </div>
  <div
    className="root"
  >
    isTight false
  </div>
  <div
    className="root roomy"
  >
    isRoomy true
  </div>
  <div
    className="root"
  >
    isRoomy false
  </div>
</div>
`;
